elixir/config/config.exs

```diff
@@ -87,6 +87,8 @@ config :domain, :enabled_features,
   self_hosted_relays: true,
   multi_site_resources: true
 
+config :domain, sign_up_allowed_domains: "firezone.dev"
```

Comment:
We should not put this as default because it will cause issues during local debugging, etc
```suggestion
config :domain, sign_up_allowed_domains: "firezone.dev"
```

Instead we should do 
```
config :domain, sign_up_allowed_domains: compile_config!(...)
```
in `runtime.exs`

Reply:
Are you saying we should add a section to the `runtime.exs` that gets executed in `:dev` mode?  There should already be:
```
config :domain, sign_up_allowed_domains: compile_config!(...)
```
in the `runtime.exs` for `:prod`

Reply:
Oh shoot, I just realized I didn't add in the change to `runtime.exs`.  I'll get that added in, but I am still curious about the suggestion for non-prod environments.

Reply:
@bmanifold for `config.exs` I would do the most permissive defaults possible (since it's just for our dev/test envs):
```
config :domain, sign_up_allowed_domains: [] # or nil, whatever means "allow all domains"
```

Reply:
Changing the default to "allow all domains" in `config.exs` makes sense.  Where I'm still not clear is if you want a section added to `runtime.exs` to evaluate when running in `:dev` mode.  At the moment, everything in `runtime.exs` appears to only be evaluated if we're running in `:prod` mode.

Reply:
@bmanifold In `dev` mode I would not set anything and use default from `config.exs`

Reply:
Sounds good 👍